producer/writer/message_writer.go

```diff
@@ -182,10 +183,10 @@ func (w *messageWriterImpl) Write(rd producer.RefCountedData) {
 	}
 	msg.Reset(meta, rd)
 	w.acks.add(meta, msg)
-	w.writeWithLock(msg, nowNanos)
 	w.queue.PushBack(msg)
+	consumerWriters := w.consumerWriters
```

Comment:
Are we sure this is not going to be mutated outside the lock?

Reply:
yes, it's only mutated within lock